### PR TITLE
[Feature] Skill summary table for RoD

### LINKED ISSUE #8127: ✨ Skill Summary Table for Assessment Page
## ✨ Feature

The assessment plan builder is the first step in the Record of decision (RoD) flow. It allows us to capture the list of assessments to be used during the staffing process, the order in which they will be performed, and the skills that each assessment will be targeting.

It features a repeater component with dialogs for details, and a table to overview how the skills and assessments are related to each other.

This issue covers just the Skill Summary Table.

## 🎨 Design File and Designer

[Link to figma file](https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Pools---Record-of-Decision?type=design&node-id=5437%3A34317&mode=design&t=11Dob6JX6BOb3lle-1) by @Jerryescandon 

## 📸 Screenshots

image

## ✅ Acceptance Criteria

- [ ] Component built in Storybook
- [ ] Design review with designer

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #8127

## 👋 Introduction

Adds a new component utilizing the responsive table.  

- [x] Design review

## 🕵️ Details

Currently in a folder for components, when the page that will use this is added, this can probably just be moved there. 

## 🧪 Testing

1. Run storybook
2. Navigate to SkillSummaryTable
3. Observe and manipulate (with controls or story file changes)

Manipulating example

https://recordit.co/iWKt8XjbJB

## 📸 Screenshot

Current state

image

### CODE REVIEW

**Comment by vd1992** [comment]
@Jerryescandon 
Ready for design review, here is a link to the story.

https://61e099a1bb1465003a73d3ce-azzkhogwlm.chromatic.com/?path=/story/components-skillsummarytable--default

File: packages/i18n/src/messages/localizedConstants.tsx
Comment (vd1992) [inline packages/i18n/src/messages/localizedConstants.tsx:1]: @Jerryescandon 

This file has the displayed copy for the enums: skill category and assessment type. 
Please glance through, these will be the constants displayed wherever the enums are called.

Comment (Jerryescandon) [reply packages/i18n/src/messages/localizedConstants.tsx:1]: Other than obviously "application screening" shouldn't be able to be edited in those ways in the real scenario 😆.

I think the structure is working and looks correct 👍

Comment (vd1992) [reply packages/i18n/src/messages/localizedConstants.tsx:1]: Excellent, thank you!

File: apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    }
+
+    return intl.formatMessage(commonMessages.notAvailable);
+  };
+
+  const plannedAssessmentCell = (poolSkill: PoolSkill): JSX.Element => {
+    return poolSkill.assessmentSteps && poolSkill.assessmentSteps.length > 0
+      ? CheckCircleIconElement
+      : XCircleIconElement;
+  };
```
Comment (esizer) [inline apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx:94]: We will need to add some text (possibly an `aria-label`) to indicate what the icons mean to AT users.

File: apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        )
+      : [];
+    const filteredPoolSkillsAssessments = unpackMaybes(
+      poolSkillsAssessmentTypes,
+    );
+    if (
+      assessmentStep.type &&
+      filteredPoolSkillsAssessments.includes(assessmentStep.type)
+    ) {
+      return CheckIconElement;
```
Comment (esizer) [inline apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx:136]: Same here, some sort of text to indicate what this icon means.

**Review by esizer — CHANGES_REQUESTED** [review]
I'm having a hard time understanding the dynamic columns and what they refer to since this isn't in context yet 😅 

But, I did notice some minor items related to a11y.

File: apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx (hunk 1 above)
Comment (vd1992) [reply apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx]: Like so https://github.com/GCTC-NTGC/gc-digital-talent/pull/8169/commits/5e83575c41edc596cf4e7c93020ca35b8da1ff6c ?

Comment (esizer) [reply apps/web/src/components/SkillSummaryTable/SkillSummaryTable.tsx]: That's amazing! ⭐ Thank-you 😄

File: packages/i18n/src/messages/localizedConstants.tsx [hunk 3]
```diff
@@ -1929,3 +1953,65 @@ export const getEvaluatedLanguageAbility = (
     ability,
     `Invalid evaluatedLanguageAbility ${ability}`,
   );
+
+const assessmentStepTypes = defineMessages({
```
Comment (petertgiles) [inline packages/i18n/src/messages/localizedConstants.tsx:1957]: Shoot, I requested the translation for these, too.  Should be interesting to see how close they come.

**Review by petertgiles — APPROVED** [review]
Nice!  Props look good to me and table seems to work well.

**Comment by Jerryescandon** [comment]
Reviewed with @petertgiles , looks good to me 👍

File: packages/i18n/src/messages/localizedConstants.tsx (hunk 3 above)
Comment (vd1992) [reply packages/i18n/src/messages/localizedConstants.tsx:1957]: Got 'em in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8169/commits/eba8d4a61ee1bbd8f0974eb253d3550f9b8eccad
Take a look if you'd like

**Pull request merged**